test/integration/kubernetes_deploy_test.rb

```diff
@@ -0,0 +1,27 @@
+require 'test_helper'
+
+class KubernetesDeployTest < KubernetesDeploy::IntegrationTest
+  def test_basic
+    runner = KubernetesDeploy::Runner.new(
+      namespace: @namespace,
+      current_sha: "abcabcabc",
+      context: "minikube",
+      template_dir: File.expand_path("./test/fixtures/basic"),
+      wait_for_completion: true,
+    )
+    runner.run
+
+    pods = kubeclient.get_pods(namespace: @namespace)
+    managed_pods, unmanaged_pods = pods.partition do |pod|
```

Comment:
This one's Kir's 😄 . I'm a fan as well though; if you wanna see more, check out `runner.rb`. 😉